x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/Evaluation.java

```diff
@@ -27,37 +34,67 @@
      */
     String getName();
 
+    /**
+     * Returns the field containing the actual value
+     */
+    String getActualField();
+
+    /**
+     * Returns the field containing the predicted value
+     */
+    String getPredictedField();
+
     /**
      * Returns the list of metrics to evaluate
      * @return list of metrics to evaluate
      */
     List<? extends EvaluationMetric> getMetrics();
 
+    default <T extends EvaluationMetric> List<T> initMetrics(@Nullable List<T> parsedMetrics, Supplier<List<T>> defaultMetricsSupplier) {
+        List<T> metrics = parsedMetrics == null ? defaultMetricsSupplier.get() : new ArrayList<>(parsedMetrics);
+        if (metrics.isEmpty()) {
+            throw ExceptionsHelper.badRequestException("[{}] must have one or more metrics", getName());
+        }
+        Collections.sort(metrics, Comparator.comparing(EvaluationMetric::getName));
+        return metrics;
+    }
+
     /**
      * Builds the search required to collect data to compute the evaluation result
      * @param userProvidedQueryBuilder User-provided query that must be respected when collecting data
      */
-    SearchSourceBuilder buildSearch(QueryBuilder userProvidedQueryBuilder);
-
-    /**
-     * Builds the search that verifies existence of required fields and applies user-provided query
-     * @param requiredFields fields that must exist
-     * @param userProvidedQueryBuilder user-provided query
-     */
-    default SearchSourceBuilder newSearchSourceBuilder(List<String> requiredFields, QueryBuilder userProvidedQueryBuilder) {
-        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
-        for (String requiredField : requiredFields) {
-            boolQuery.filter(QueryBuilders.existsQuery(requiredField));
+    default SearchSourceBuilder buildSearch(QueryBuilder userProvidedQueryBuilder) {
+        ExceptionsHelper.requireNonNull(userProvidedQueryBuilder, "userProvidedQueryBuilder");
```

Comment:
You can use `Objects.requireNonNull` here as this is the query is not provided directly from the user but from our code instead.

Reply:
Done.